peer/internal/uprng/uprng.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2024 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package uprng
+
+import (
+	cryptorand "crypto/rand"
+	"io"
+	"sync"
+	"time"
+
+	"golang.org/x/crypto/chacha20"
+)
+
+// Reader implements a cryptographically secure userspace PRNG that is
+// periodically reseeded with entropy obtained from crypto/rand.
+// Reader is safe for concurrent access.
+var Reader io.Reader
+
+func init() {
+	Reader = new(prng)
```

Comment:
No need for this to be an init func.  Just do `var Reader io.Reader = new(prng)`.

Reply:
I know it's not needed, but I don't like assigning exported variables to instances of unexported types because they show up in the godoc.

```
$ go doc uprng.Reader
package uprng // import "github.com/decred/dcrd/peer/internal/uprng"

var Reader io.Reader = new(prng)
    Reader implements a cryptographically secure userspace PRNG that is
    periodically reseeded with entropy obtained from crypto/rand. Reader is safe
    for concurrent access.

```

Reply:
Fair enough.  That doesn't bother me personally.  I find it easier when reading the code to see what it is instead of chasing down inits in general, but not a big deal.  I'm fine either way.